modules/claims_api/lib/claims_api/v2/disability_compensation_pdf_mapper.rb

```diff
@@ -763,19 +785,34 @@ def additional_identification_info
 
       def regex_date_conversion(date)
         if date.present?
-          res = date.match(/^(?:(?<year>\d{4})(?:-(?<month>\d{2}))?(?:-(?<day>\d{2}))*|(?<month>\d{2})?(?:-(?<day>\d{2}))?-?(?<year>\d{4}))$/) # rubocop:disable Layout/LineLength
-
-          make_date_object(res, date.length)
+          date_match = date.match(/^(?:(?<year>\d{4})(?:-(?<month>\d{2}))?(?:-(?<day>\d{2}))*|(?<month>\d{2})?(?:-(?<day>\d{2}))?-?(?<year>\d{4}))$/) # rubocop:disable Layout/LineLength
+          date_match&.values_at(:year, :month, :day)
         end
       end
 
       def make_date_object(date, date_length)
-        if date.present? && date_length == 4
-          { year: date[:year] }
-        elsif date.present? && date_length == 7
-          { month: date[:month], year: date[:year] }
-        elsif date.present?
-          { year: date[:year], month: date[:month], day: date[:day] }
+        year, month, day = regex_date_conversion(date)
+        return if year.nil?
+
+        if date_length == 4
+          { year: }
+        elsif date_length == 7
+          { month:, year: }
+        else
+          { year:, month:, day: }
```

Comment:
@acovrig had a great idea to just return this with a `.compact` to remove all of the different `if's`. if it's nil it will remove them.